components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/SelfRegistrationUtils.java

```diff
@@ -0,0 +1,338 @@
+        }
+        try {
+            userStoreManager.setUserClaimValues(userName, userClaims, null);
+            if (LOG.isDebugEnabled()) {
+                if (isAccountLockOnCreation) {
+                    LOG.debug("Locked user account: " + maskIfRequired(userName));
+                }
+                if (isEnableConfirmationOnCreation) {
+                    LOG.debug("Send verification notification for user account: " + maskIfRequired(userName));
+                }
+            }
+        } catch (org.wso2.carbon.user.core.UserStoreException e) {
+            throw new IdentityEventException("Error while lock user account :" + maskIfRequired(userName), e);
+        }
+    }
+
+    /**
+     * Resolve the event name according to the notification channel.
+     *
+     * @param preferredChannel User preferred notification channel
+     * @param userName         Username
+     * @param domainName       Domain name
+     * @param tenantDomain     Tenant domain name
+     * @return Resolved event name
+     */
+    public static String resolveEventName(String preferredChannel, String userName, String domainName,
+                                          String tenantDomain) {
+
+        String eventName;
+        if (NotificationChannels.EMAIL_CHANNEL.getChannelType().equals(preferredChannel)) {
+            eventName = IdentityEventConstants.Event.TRIGGER_NOTIFICATION;
+        } else {
+            eventName = IdentityRecoveryConstants.NOTIFICATION_EVENTNAME_PREFIX + preferredChannel
+                    + IdentityRecoveryConstants.NOTIFICATION_EVENTNAME_SUFFIX;
+        }
+        if (LOG.isDebugEnabled()) {
+            String message = String
+                    .format("For user : %1$s in domain : %2$s, notifications were sent from the event : %3$s",
+                            maskIfRequired(domainName + DOMAIN_SEPARATOR + userName), tenantDomain,
+                            eventName);
+            LOG.debug(message);
+        }
+        return eventName;
+    }
+
+    /**
+     * Triggers notifications according to the given event name.
+     *
+     * @param user                User
+     * @param notificationChannel Notification channel
+     * @param code                Recovery code
+     * @param props               Event properties
+     * @param eventName           Name of the event
+     * @throws IdentityRecoveryException Error triggering notifications
+     */
+    public static void triggerNotification(User user, String notificationChannel, String code, Property[] props,
+                                           String eventName) throws IdentityRecoveryException, IdentityEventException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Sending self user registration notification user: " + maskIfRequired(user.getUserName()));
+        }
+
+        String serviceProviderUUID = (String) IdentityUtil.threadLocalProperties.get().get(SERVICE_PROVIDER_UUID);
```

Comment:
was this property already available

Reply:
This property is set from the old SelfRegistrationManager. If that value is not set, that property is ignored.